packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -1089,72 +1089,68 @@ export class WorkflowExecute {
 		return customOperation;
 	}
 
-	/** Executes the given node */
-	// eslint-disable-next-line complexity
-	async runNode(
-		workflow: Workflow,
-		executionData: IExecuteData,
-		runExecutionData: IRunExecutionData,
-		runIndex: number,
-		additionalData: IWorkflowExecuteAdditionalData,
-		mode: WorkflowExecuteMode,
-		abortSignal?: AbortSignal,
-	): Promise<IRunNodeResponse> {
-		const { node } = executionData;
-		let inputData = executionData.data;
-
-		if (node.disabled === true) {
-			// If node is disabled simply pass the data through
-			// return NodeRunHelpers.
-			if (inputData.hasOwnProperty('main') && inputData.main.length > 0) {
-				// If the node is disabled simply return the data from the first main input
-				if (inputData.main[0] === null) {
-					return { data: undefined };
-				}
-				return { data: [inputData.main[0]] };
+	/**
+	 * Handles execution of disabled nodes by passing through input data
+	 */
+	private handleDisabledNode(inputData: ITaskDataConnections): IRunNodeResponse {
+		if (inputData.hasOwnProperty('main') && inputData.main.length > 0) {
+			// If the node is disabled simply return the data from the first main input
+			if (inputData.main[0] === null) {
+				return { data: undefined };
 			}
-			return { data: undefined };
+			return { data: [inputData.main[0]] };
 		}
+		return { data: undefined };
+	}
 
-		const nodeType = workflow.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
-
-		const isDeclarativeNode = nodeType.description.requestDefaults !== undefined;
-
-		const customOperation = this.getCustomOperation(node, nodeType);
-
-		let connectionInputData: INodeExecutionData[] = [];
+	/**
+	 * Prepares input data for node execution based on node type and workflow settings
+	 */
+	private prepareConnectionInputData(
+		workflow: Workflow,
+		nodeType: INodeType,
+		customOperation: ReturnType<WorkflowExecute['getCustomOperation']>,
+		inputData: ITaskDataConnections,
+	): INodeExecutionData[] | null {
 		if (
-			nodeType.execute ||
-			customOperation ||
-			(!nodeType.poll && !nodeType.trigger && !nodeType.webhook)
+			!nodeType.execute &&
+			!customOperation &&
+			(nodeType.poll || nodeType.trigger || nodeType.webhook)
```

Comment:
I found this condition a bit easier to read beforehand.

Curious why we need to check that a node type does not have an `execute` method and does not have a custom node operation _in addition to_ checking that it has a method that identifies it as a trigger, poller or webhook? I'd expect the latter should be enough?